### PR TITLE
[C-2101] Improve initial download ui performance

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

Majorly improves initial download ui performance by ensuring download loading status starts right when user presses toggle.

- improves favorite download components and control flow
- fixes bug with download removal reducer

Will address the performance of collection header ui separately, but will use same techniques

### CODE REVIEW

File: packages/mobile/src/screens/favorites-screen/FavoritesDownloadStatusIndicator.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import { OfflineDownloadStatus } from 'app/store/offline-downloads/slice'
+import { makeStyles } from 'app/styles'
+
+const useStyles = makeStyles(({ spacing }) => ({
+  root: {
+    marginRight: spacing(1)
+  }
+}))
+
+const getFavoritesDownloadStatus = (state: AppState) => {
```
Comment (sliptype) [inline packages/mobile/src/screens/favorites-screen/FavoritesDownloadStatusIndicator.tsx:15]: Big fan of this complex selector pattern

File: packages/mobile/src/screens/favorites-screen/FavoritesDownloadSection.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+import { DownloadFavoritesSwitch } from './DownloadFavoritesSwitch'
+import { DownloadProgress } from './DownloadProgress'
+import { FavoritesDownloadStatusIndicator } from './FavoritesDownloadStatusIndicator'
+
+const useStyles = makeStyles(() => ({
+  root: { flexDirection: 'row', alignItems: 'center' }
+}))
+
+export const FavoritesDownloadSection = () => {
```
Comment (sliptype) [inline packages/mobile/src/screens/favorites-screen/FavoritesDownloadSection.tsx:15]: Awesome job breaking the component into smaller pieces! I wonder if there's a more descriptive name for this one than `Section`, I can't think of one though lol

**Review by sliptype — APPROVED** [review]
Great work dude! Much easier to understand now 🔥

File: packages/mobile/src/screens/favorites-screen/FavoritesDownloadSection.tsx (hunk 2 above)
Comment (dylanjeffers) [reply packages/mobile/src/screens/favorites-screen/FavoritesDownloadSection.tsx:15]: Me too! I tried for a while and then just went with this... like header maybe?

**Pull request merged**